postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py

```diff
@@ -0,0 +1,382 @@
+    xyz = (1./3.)*(geom[connect[:,0],:]+geom[connect[:,1],:]+geom[connect[:,2],:])   
+    return xyz
+
+def CalculateSlipCentroid(faultxdmf, events=0):
+    xyz = get_xyz_from_connect(faultxdmf.ReadGeometry(), faultxdmf.ReadConnect())
+    ASl = faultxdmf.ReadData("ASl", idt=lastTimeStep).T
+    if events == 0:
+        ASl = faultxdmf.ReadData("ASl", idt=lastTimeStep).T
+        xyzc = np.average(xyz, axis=0, weights=ASl)
+    elif events == 1:
+        ASl = faultxdmf.ReadData("ASl", idt=int(lastTimeStep/2)).T
+        xyzc = np.average(xyz, axis=0, weights=ASl)    
+    else:
+        ASl = faultxdmf.ReadData("ASl", idt=lastTimeStep).T - faultxdmf.ReadData("ASl", idt=int(lastTimeStep/2)).T
+        xyzc = np.average(xyz, axis=0, weights=ASl)  
+    return xyzc
+
+def ComputeBackazimuth(xyz, centroid):
+    ba = np.arctan2(xyz[:,0]-centroid[0], xyz[:,1]-centroid[1]) * 180 / np.pi % 360
+    ba = np.where(ba < 180, ba+180, ba-180)
+    return ba
+
+def RotateToRadialTransversal(v, u, xyz, centroid):
+    ba = ComputeBackazimuth(xyz, centroid)
+    ba = np.radians(ba).reshape(-1,1)
+    r = -u * np.sin(ba) - v * np.cos(ba)
+    t = -u * np.cos(ba) + v * np.sin(ba)
+    return r, t
+
+def ComputeCornerFrequencyParallel(component, pphases, windowduration):
+    chunk_inds = np.int_(np.linspace(0, component.shape[0], nprocs+1))
+    chunks = []
+    for i in range(nprocs):
+        chunks.append([component[chunk_inds[i]:chunk_inds[i+1]], pphases[chunk_inds[i]:chunk_inds[i+1]], 
+                     windowduration[chunk_inds[i]:chunk_inds[i+1]]])
+    chunks = [tuple(i) for i in chunks]  
+       
+    p = Pool(nprocs)
+    result = p.map(ComputeCornerFrequencyArray, chunks)
+    p.close()
+    p.join()
+    return np.concatenate(result)        
+
+def ComputeCornerFrequencyArray(tuples):
+    fc_arr = np.zeros(tuples[0].shape[0])
+    for i in range(0, fc_arr.size):
+        fc_arr[i] = ComputeSingleCornerFrequency(tuples[0][i], dt, pphase=tuples[1][i], 
+                                                 windowduration=tuples[2][i], endfreq=args.maxFreq[0])
+    return fc_arr
+
+def ComputeSingleCornerFrequency(velocity, dt, pphase=0, windowduration=0, endfreq=1.0):
+
+    # preprocessing (taper and padding)
+    if windowduration != 0:
+        data = cut_window_and_pad(velocity, windowduration, dt, pphase=pphase, alpha=0.05)
+    else:
+        data = velocity
+    
+    # fourier transform and down trimming to maxFreq
+    data, freqs = complete_fft(data, dt)
+    maxFreqIndices = int(np.argwhere(freqs > endfreq)[1])
+    data = data[1:maxFreqIndices]
+    freqs = freqs[1:maxFreqIndices]
+    
+    # integrate within the frequency domain and convert to amplitude spectrum
+    data = np.abs(data*(1/(1j*2*np.pi*freqs)))
+    
+    newfreqs = np.arange(0.01, endfreq+0.005, 0.005)
+    data = np.interp(newfreqs, freqs, data)
+    
+    fc, q = fc_Qzero_gridsearch(data, newfreqs, misfit="SSM", n=2, fc_end=args.maxCornerFreq[0])
```

Comment:
why to return q if you are not using it elsewhere in the code?

Reply:
Changed.